### PR TITLE
Update to `ndx_fiber_photometry`

### PR DESCRIPTION
**Pull request by weiglszonja**

### CODE REVIEW

**Comment by weiglszonja** [comment]
Requires https://github.com/catalystneuro/ndx-fiber-photometry/pull/4

File: src/dombeck_lab_to_nwb/azcorra2023/azcorra2023_requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 ndx-events @ git+https://github.com/rly/ndx-events.git # Until 0.3.0 is released
-ndx-photometry @ git+https://github.com/catalystneuro/ndx-photometry.git@main # Until new release (0.4.0) is made
+ndx-fiber-photometry @ git+https://github.com/catalystneuro/ndx-fiber-photometry.git@create_spec
```
Comment (weiglszonja) [inline src/dombeck_lab_to_nwb/azcorra2023/azcorra2023_requirements.txt:2]:
```suggestion
ndx-fiber-photometry @ git+https://github.com/catalystneuro/ndx-fiber-photometry.git@main
```

File: src/dombeck_lab_to_nwb/azcorra2023/photometry_utils/add_fiber_photometry.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    photometry_device = defaultdict(
+        OpticalFiber=OpticalFiber,
+        Indicator=Indicator,
+        ExcitationSource=ExcitationSource,
+        Photodetector=Photodetector,
+        DichroicMirror=DichroicMirror,
+        OpticalFilter=OpticalFilter,
+    )[device_type](**device_metadata)
+
+    nwbfile.add_device(photometry_device)
```
Comment (weiglszonja) [inline src/dombeck_lab_to_nwb/azcorra2023/photometry_utils/add_fiber_photometry.py:37]: @pauladkisson @luiztauffer This has really simplified creating the "devices" from the metadata for me. Maybe will be useful also for TDT fiber photometry interface on `neuroconv`?

**Comment by weiglszonja** [comment]
Stub [file](https://us-east-2.console.aws.amazon.com/s3/object/catalystneuro-processing?region=us-east-2&bucketType=general&prefix=dombeck-lab-to-nwb/t20200129-0002.nwb) for discussion

File: src/dombeck_lab_to_nwb/azcorra2023/photometry_utils/add_fiber_photometry.py (hunk 2 above)
Comment (pauladkisson) [reply src/dombeck_lab_to_nwb/azcorra2023/photometry_utils/add_fiber_photometry.py:37]: Thanks szonja! I'll try it out.

File: src/dombeck_lab_to_nwb/azcorra2023/metadata/azcorra2023_fiber_photometry_metadata.yaml [hunk 3]
```diff
@@ -1,82 +1,120 @@
...
+        model: FF01-406/15-25
+        peak_wavelength_in_nm: 406.0
+        bandwidth_in_nm: [391.0, 431.0]
+        filter_type: Excitation  # TODO: not sure where to find the emission filter parameters (see: https://www.protocols.io/view/acute-striatal-or-midbrain-fiber-photometry-in-hea-4r3l27yj4g1y/v1?step=29)
+      - name: OpticalFilter470
+        description: The filter used to isolate the 470 nm excitation light.
+        manufacturer: Semrock
+        model: FF02-472/30-25
+        peak_wavelength_in_nm: 472.0
+        bandwidth_in_nm: [442.0, 497.0]
```
Comment (weiglszonja) [inline src/dombeck_lab_to_nwb/azcorra2023/metadata/azcorra2023_fiber_photometry_metadata.yaml:96]:
```suggestion
        bandwidth_in_nm: [457.5 , 487.5]
```

Comment (weiglszonja) [reply src/dombeck_lab_to_nwb/azcorra2023/metadata/azcorra2023_fiber_photometry_metadata.yaml:96]: + we can add description the band was taken from data sheet

File: src/dombeck_lab_to_nwb/azcorra2023/metadata/azcorra2023_fiber_photometry_metadata.yaml [hunk 4]
```diff
@@ -1,82 +1,120 @@
...
+        filter_type: Excitation
```
Comment (weiglszonja) [inline src/dombeck_lab_to_nwb/azcorra2023/metadata/azcorra2023_fiber_photometry_metadata.yaml:97]:
```suggestion
        filter_type: Bandpass
```

**Comment by weiglszonja** [comment]
@alessandratrapani ready for review

File: src/dombeck_lab_to_nwb/azcorra2023/photometry_utils/add_fiber_photometry.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        excitation_filter=nwbfile.devices[optical_filter_to_add],
+        emission_filter=nwbfile.devices[emission_filter_to_add],
+    )
+
+    fiber_photometry_table_region = DynamicTableRegion(
+        name="fiber_photometry_table_region",
+        data=[table_region_ind],
+        description="source fibers",
+        table=fiber_photometry_table,
+    )
```
Comment (weiglszonja) [inline src/dombeck_lab_to_nwb/azcorra2023/photometry_utils/add_fiber_photometry.py:177]:
```suggestion
    fiber_photometry_table_region = fiber_photometry_table.create_fiber_photometry_table_region(
            region=[table_region_ind], description="source fibers"
        )
```

**Review by alessandratrapani — APPROVED** [review]
LOOKS VERY GOOD!

**Pull request merged**